orc/src/main/java/org/apache/iceberg/orc/ORC.java

```diff
@@ -177,9 +179,20 @@ public ReadBuilder filter(Expression newFilter) {
       return this;
     }
 
+    public ReadBuilder createBatchedReaderFunc(Function<TypeDescription, OrcBatchReader<?>> batchReaderFunction) {
+      this.batchedReaderFunc = batchReaderFunction;
```

Comment:
This should validate that `createReaderFunc` hasn't also been called. And that function should validate that this one hasn't been called. That way there is no ambiguous behavior and the user gets an error when they try to set up both batch and row readers.

Reply:
Fixed in https://github.com/apache/iceberg/pull/1189/commits/d13c3662ac5d7417c647ad5c6c5b22685341dc46